### ISSUE TITLE
Not OK: perl v5.8.1 +MAINT19791 on MSWin32-x86-multi-thread 4.0 (UNINSTALLED)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6572]
Migrated from [rt.perl.org#22728](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=22728) (status was 'resolved')

Searchable as RT22728$

**Comment by p5pRT** [comment]
### From andreas@mrs.ch



Just in case anybody is interested already \(don't know when the next
release is due\)\.

MingW32\, WinNT6\, SP6a

Failed Test                      Stat Wstat Total Fail  Failed  List of Failed
\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\.\./ext/Devel/DProf/DProf\.t                     18   13  72\.22%  1\-8 13\-17
\.\./lib/Net/Ping/t/450\_service\.t                26    4  15\.38%  8\-10 22
\.\./lib/Net/Ping/t/510\_ping\_udp\.t                2    1  50\.00%  2
io/dup\.t                                       12    2  16\.67%  6\-7
io/open\.t                                     100    2   2\.00%  35\-36
io/openpid\.t                                   10    1  10\.00%  9
win32/system\.t                               1008   12   1\.19%  225 253 281 309
&nbsp;                                                                561\-562 589
&nbsp;                                                                617\-618 645 785
&nbsp;                                                                813
 \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
dmake\.exe&#8203;:  Error code 255\, while making 'test'

<details><summary>Perl Info</summary>

```

Flags:
    category=install
    severity=none

Site configuration information for perl v5.8.1:

Configured by FehrA at Wed Jun 18 08:18:34 2003.

Summary of my perl5 (revision 5 version 8 subversion 1) configuration:
  Platform:
    osname=MSWin32, osvers=4.0, archname=MSWin32-x86-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-g -O2 -DWIN32 -DHAVE_DES_FCRYPT -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -fno-strict-aliasing -DPERL_MSVCRT_READFIX',
    optimize='-g -O2',
    cppflags='-DWIN32'
    ccversion='', gccversion='', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='long long', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-g -L"c:\bin\temp\perl\5.9.0\lib\MSWin32-x86-multi-thread\CORE" -L"c:\bin\fsf\MingW32\lib"'
    libpth=c:\bin\fsf\MingW32\lib
    libs= -lmsvcrt -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lwsock32 -lmpr -lwinmm -lversion -lodbc32
    perllibs= -lmsvcrt -lmoldname -lkernel32 -luser32 -lgdi32 -lwinspool -lcomdlg32 -ladvapi32 -lshell32 -lole32 -loleaut32 -lnetapi32 -luuid -lwsock32 -lmpr -lwinmm -lversion -lodbc32
    libc=-lmsvcrt, so=dll, useshrplib=yes, libperl=libperl58.a
    gnulibc_version='undef'
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-mdll -g -L"c:\bin\temp\perl\5.9.0\lib\MSWin32-x86-multi-thread\CORE" -L"c:\bin\fsf\MingW32\lib"'

Locally applied patches:
    MAINT19791


@INC for perl v5.8.1:
    ..\lib
    c:/docs/source/perl
    F:/temp/perl@19792/lib
    .


Environment for perl v5.8.1:
    HOME=M:\
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=C
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=C:\bin\fsf\MingW32\bin;C:\WINNT4.0\system32;C:\WINNT4.0;C:\bin\perl\5.6.1\bin;C:\bin\perl\5.6.1\bin\MSWin32-x86-multi-thread;c:\bin\lonworks\bin;C:\bin\fsf\NcFTP;c:\bin\exe;c:\bin\fsf\vim\vim61;c:\bin\fsf\sshd\ssh;C:\bin\fsf\NCFTP-~1.5;c:\bin\fsf\dmake
    PERL5LIB=c:/docs/source/perl
    PERL_BADLANG (unset)
    SHELL (unset)


-- 
Andreas Fehr
andreas@mrs.ch

```
</details>

**Comment by p5pRT** [comment]
### From @rgs

Andreas Fehr wrote&#8203;:
> 
> MingW32\, WinNT6\, SP6a
> 
> Failed Test                      Stat Wstat Total Fail  Failed  List of Failed
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \.\./ext/Devel/DProf/DProf\.t                     18   13  72\.22%  1\-8 13\-17
> \.\./lib/Net/Ping/t/450\_service\.t                26    4  15\.38%  8\-10 22
> \.\./lib/Net/Ping/t/510\_ping\_udp\.t                2    1  50\.00%  2
> io/dup\.t                                       12    2  16\.67%  6\-7
> io/open\.t                                     100    2   2\.00%  35\-36
> io/openpid\.t                                   10    1  10\.00%  9
> win32/system\.t                               1008   12   1\.19%  225 253 281 309
>                                                                 561\-562 589
>                                                                 617\-618 645 785
>                                                                 813
>  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
> Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
> dmake\.exe&#8203;:  Error code 255\, while making 'test'

Let's see the bright side of things :
Any way to know which test UNEXPECTEDLY SUCCEEDED ?

**Comment by p5pRT** [comment]
### From andreas@mrs.ch

On Thu\, 18 Jun 2003\, 22&#8203;:06\, Rafael Garcia\-Suarez \(as RG\) wrote&#8203;:

>RG&#8203;: Andreas Fehr wrote&#8203;:
>RG&#8203;: >
>RG&#8203;: > MingW32\, WinNT6\, SP6a
>RG&#8203;: >
>RG&#8203;: > Failed Test                      Stat Wstat Total Fail  Failed  List of Failed
>RG&#8203;: > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-

\<snip>

>RG&#8203;: >  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
>RG&#8203;: > Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
>RG&#8203;: > dmake\.exe&#8203;:  Error code 255\, while making 'test'
>RG&#8203;:
>RG&#8203;: Let's see the bright side of things :
>RG&#8203;: Any way to know which test UNEXPECTEDLY SUCCEEDED ?

Sorry\, no help from my side\. I did run the test again\, but was not able
the find it\. Any idea how to find this?

**Comment by p5pRT** [comment]
### From @rgs

Andreas Fehr \<andreas@&#8203;mrs\.ch> wrote&#8203;:
> >RG&#8203;: >  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
> >RG&#8203;: > Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
> >RG&#8203;: > dmake\.exe&#8203;:  Error code 255\, while making 'test'
> >RG&#8203;:
> >RG&#8203;: Let's see the bright side of things :
> >RG&#8203;: Any way to know which test UNEXPECTEDLY SUCCEEDED ?
> 
> Sorry\, no help from my side\. I did run the test again\, but was not able
> the find it\. Any idea how to find this?

You can run the tests with the environment variable HARNESS\_VERBOSE=1\,
and search the resulting \(bigger\) logs for /^ok\.\*TODO/\.

**Comment by p5pRT** [comment]
### From @schwern

On Mon\, Jun 23\, 2003 at 12&#8203;:06&#8203;:40PM \+0200\, Rafael Garcia\-Suarez wrote&#8203;:
> Andreas Fehr \<andreas@&#8203;mrs\.ch> wrote&#8203;:
> > >RG&#8203;: >  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
> > >RG&#8203;: > Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
> > >RG&#8203;: > dmake\.exe&#8203;:  Error code 255\, while making 'test'
> > >RG&#8203;:
> > >RG&#8203;: Let's see the bright side of things :
> > >RG&#8203;: Any way to know which test UNEXPECTEDLY SUCCEEDED ?
> > 
> > Sorry\, no help from my side\. I did run the test again\, but was not able
> > the find it\. Any idea how to find this?
> 
> You can run the tests with the environment variable HARNESS\_VERBOSE=1\,
> and search the resulting \(bigger\) logs for /^ok\.\*TODO/\.

Or you can run 'make test\_harness'\.  Or you can edit t/TEST a bit like so&#8203;:

&nbsp;                        if\( $not && \!$istodo \) \{
&nbsp;                            $ok = 0;
&nbsp;                            $next = $num;
&nbsp;                            last;
&nbsp;                        \}
&nbsp;                        else \{
&nbsp;                            print STDERR "\\n$num UNEXPECTEDLY SUCCEEDED\\n"
&nbsp;                if $istodo;
&nbsp;                            $next = $next \+ 1;
&nbsp;                        \}


\-\- 
Hey you kids\, come back here\!  I know who your parents are\!  You wouldn't
do this if Nixon was in the White House\!  Come on\, I'm the Walrus\, damnit\!

**Comment by p5pRT** [comment]
### From andreas@mrs.ch

On Mon\, 23 Jun 2003\, 03&#8203;:33\, Michael G Schwern \(as MGS\) wrote&#8203;:

>MGS&#8203;: On Mon\, Jun 23\, 2003 at 12&#8203;:06&#8203;:40PM \+0200\, Rafael Garcia\-Suarez wrote&#8203;:
>MGS&#8203;: > Andreas Fehr \<andreas> wrote&#8203;:
>MGS&#8203;: > > >RG&#8203;: >  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
>MGS&#8203;: > > >RG&#8203;: > Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
>MGS&#8203;: > > >RG&#8203;: > dmake\.exe&#8203;:  Error code 255\, while making 'test'
>MGS&#8203;: > > >RG&#8203;:
>MGS&#8203;: > > >RG&#8203;: Let's see the bright side of things :
>MGS&#8203;: > > >RG&#8203;: Any way to know which test UNEXPECTEDLY SUCCEEDED ?
>MGS&#8203;: > >
>MGS&#8203;: > > Sorry\, no help from my side\. I did run the test again\, but was not able
>MGS&#8203;: > > the find it\. Any idea how to find this?
>MGS&#8203;: >
>MGS&#8203;: > You can run the tests with the environment variable HARNESS\_VERBOSE=1\,
>MGS&#8203;: > and search the resulting \(bigger\) logs for /^ok\.\*TODO/\.
>MGS&#8203;:
>MGS&#8203;: Or you can run 'make test\_harness'\.

"Don't know how to make test\_harness"



>MGS&#8203;:                                       Or you can edit t/TEST a bit like so&#8203;:
>MGS&#8203;:
>MGS&#8203;:                         if\( $not && \!$istodo \) \{
>MGS&#8203;:                             $ok = 0;
>MGS&#8203;:                             $next = $num;
>MGS&#8203;:                             last;
>MGS&#8203;:                         \}
>MGS&#8203;:                         else \{
>MGS&#8203;:                             print STDERR "\\n$num UNEXPECTEDLY SUCCEEDED\\n"
>MGS&#8203;:                 if $istodo;
>MGS&#8203;:                             $next = $next \+ 1;
>MGS&#8203;:                         \}

Might work\, but did it with Rafaels hint\.

Andreas

**Comment by p5pRT** [comment]
### From andreas@mrs.ch

On Mon\, 23 Jun 2003\, 12&#8203;:06\, Rafael Garcia\-Suarez \(as RG\) wrote&#8203;:

>RG&#8203;: Andreas Fehr \<andreas> wrote&#8203;:
>RG&#8203;: > >RG&#8203;: >  \(1 subtest UNEXPECTEDLY SUCCEEDED\)\, 55 tests and 665 subtests skipped\.
>RG&#8203;: > >RG&#8203;: > Failed 7/794 test scripts\, 99\.12% okay\. 35/75119 subtests failed\, 99\.95% okay\.
>RG&#8203;: > >RG&#8203;: > dmake\.exe&#8203;:  Error code 255\, while making 'test'
>RG&#8203;: > >RG&#8203;:
>RG&#8203;: > >RG&#8203;: Let's see the bright side of things :
>RG&#8203;: > >RG&#8203;: Any way to know which test UNEXPECTEDLY SUCCEEDED ?
>RG&#8203;: >
>RG&#8203;: > Sorry\, no help from my side\. I did run the test again\, but was not able
>RG&#8203;: > the find it\. Any idea how to find this?
>RG&#8203;:
>RG&#8203;: You can run the tests with the environment variable HARNESS\_VERBOSE=1\,
>RG&#8203;: and search the resulting \(bigger\) logs for /^ok\.\*TODO/\.
>RG&#8203;:

Does this help&#8203;:

\.\./lib/Net/Ping/t/450\_service\.\.\.\.\.\.\.\.\.\.1\.\.26 todo 18;
\# Running under perl version 5\.008001 for MSWin32
\# Current time local&#8203;: Mon Jun 23 14&#8203;:35&#8203;:55 2003
\# Current time GMT&#8203;:   Mon Jun 23 12&#8203;:35&#8203;:55 2003
\# Using Test\.pm version 1\.24
ok 1
ok 2
ok 3
ok 4
ok 5
ok 6
ok 7
not ok 8
not ok 9
not ok 10
ok 11
ok 12
ok 13
ok 14
ok 15
ok 16
ok 17
ok 18 \# \(\.\./lib/Net/Ping/t/450\_service\.t at line 142 TODO?\!\)
ok 19
ok 20
ok 21
not ok 22
ok 23
ok 24
ok 25
ok 26
FAILED tests 8\-10\, 22

&nbsp;    Failed 4/26 tests\, 84\.62% okay
\.\./lib/Test/Harness/t/strap\.\.\.\.\.\.\.\.\.\.\.\.1\.\.147
ok 1 \- use Test&#8203;::Harness&#8203;::Straps;
ok 2 \- new\(\)
ok 3 \- \_is\_comment\(\)\, not a comment
ok 4 \-   no comment set
ok 5 \-   not a comment with \\\#
ok 6 \-   no comment set
ok 7 \-   comment '\\\#'
ok 8 \-   right comment set
ok 9 \-   comment '    \\\# more things '
ok 10 \-   right comment set
ok 11 \-   comment '\\\# stuff and things \\\#'
ok 12 \-   right comment set
ok 13 \- \_is\_header\(\)\, not a header ' 1\.\.2'
ok 14 \-   max\, todo and skip\_all are not set
ok 15 \- \_is\_header\(\)\, not a header '1\.\.M'
ok 16 \-   max\, todo and skip\_all are not set
ok 17 \- \_is\_header\(\)\, not a header '1\.\.\-1'
ok 18 \-   max\, todo and skip\_all are not set
ok 19 \- \_is\_header\(\)\, not a header '2\.\.2'
ok 20 \-   max\, todo and skip\_all are not set
ok 21 \- \_is\_header\(\)\, not a header '1\.\.a'
ok 22 \-   max\, todo and skip\_all are not set
ok 23 \- \_is\_header\(\)\, not a header ''
ok 24 \-   max\, todo and skip\_all are not set
ok 25 \- \_is\_header\(\) is a header '1\.\.0 \\\# skip skip skip because'
ok 26 \-   skip\_all set right
ok 27 \-   the right attributes are there
ok 28 \- \_is\_header\(\) is a header '1\.\.1'
ok 29 \-   the right attributes are there
ok 30 \- \_is\_header\(\) is a header '1\.\.0 \\\# Skipped&#8203;: no leverage found'
ok 31 \-   skip\_all set right
ok 32 \-   the right attributes are there
ok 33 \- \_is\_header\(\) is a header '1\.\.4 \\\# Skipped&#8203;: no leverage found'
ok 34 \-   skip\_all set right
ok 35 \-   the right attributes are there
ok 36 \- \_is\_header\(\) is a header '1\.\.192 todo 4 2 13 192 \\\# Skip skip skip because'
ok 37 \-   skip\_all set right
ok 38 \-   the right attributes are there
ok 39 \- \_is\_header\(\) is a header '1\.\.2'
ok 40 \-   the right attributes are there
ok 41 \- \_is\_header\(\) is a header '1\.\.10 todo 2 4 10'
ok 42 \-   the right attributes are there
ok 43 \- \_is\_header\(\) is a header '1\.\.0'
ok 44 \-   skip\_all set right
ok 45 \-   the right attributes are there
ok 46 \- \_is\_header\(\) is a header '1\.\.10 todo'
ok 47 \-   the right attributes are there
ok 48 \- \_is\_test\(\) spots 'not ok'
ok 49 \-   ok
ok 50 \-   number
ok 51 \-   name
ok 52 \-   type
ok 53 \-   reason
ok 54 \- \_is\_test\(\) spots 'ok'
ok 55 \-   ok
ok 56 \-   number
ok 57 \-   name
ok 58 \-   type
ok 59 \-   reason
ok 60 \- \_is\_test\(\) spots 'not ok 23 \\\# TODO world peace'
ok 61 \-   ok
ok 62 \-   number
ok 63 \-   name
ok 64 \-   type
ok 65 \-   reason
ok 66 \- \_is\_test\(\) spots 'not ok 1'
ok 67 \-   ok
ok 68 \-   number
ok 69 \-   name
ok 70 \-   type
ok 71 \-   reason
ok 72 \- \_is\_test\(\) spots 'not ok 11 \- this is \\\\\# all the name \\\# skip this is not'
ok 73 \-   ok
ok 74 \-   number
ok 75 \-   name
ok 76 \-   type
ok 77 \-   reason
ok 78 \- \_is\_test\(\) spots 'ok \\\# skip'
ok 79 \-   ok
ok 80 \-   number
ok 81 \-   name
ok 82 \-   type
ok 83 \-   reason
ok 84 \- \_is\_test\(\) spots 'not ok 42 \- universal constant'
ok 85 \-   ok
ok 86 \-   number
ok 87 \-   name
ok 88 \-   type
ok 89 \-   reason
ok 90 \- \_is\_test\(\) spots 'ok 42 \- \_is\_header\(\) is a header '1\.\.192 todo 4 2 13 192 \\\\\# Skip skip skip because'
ok 91 \-   ok
ok 92 \-   number
ok 93 \-   name
ok 94 \-   type
ok 95 \-   reason
ok 96 \- \_is\_test\(\) spots 'ok 1'
ok 97 \-   ok
ok 98 \-   number
ok 99 \-   name
ok 100 \-   type
ok 101 \-   reason
ok 102 \- \_is\_test\(\) spots 'not ok \\\# TODO'
ok 103 \-   ok
ok 104 \-   number
ok 105 \-   name
ok 106 \-   type
ok 107 \-   reason
ok 108 \- \_is\_test\(\) spots 'ok 11 \- have life \\\# TODO get a life'
ok 109 \-   ok
ok 110 \-   number
ok 111 \-   name
ok 112 \-   type
ok 113 \-   reason
ok 114 \- \_is\_test\(\) spots 'ok 1066 \- and all that'
ok 115 \-   ok
ok 116 \-   number
ok 117 \-   name
ok 118 \-   type
ok 119 \-   reason
ok 120 \- \_is\_test\(\) spots 'ok 2938'
ok 121 \-   ok
ok 122 \-   number
ok 123 \-   name
ok 124 \-   type
ok 125 \-   reason
ok 126 \- \_is\_test\(\) disregards ' ok'
ok 127 \-   and produces no test info
ok 128 \- \_is\_test\(\) disregards 'not'
ok 129 \-   and produces no test info
ok 130 \- \_is\_test\(\) disregards 'okay 23'
ok 131 \-   and produces no test info
ok 132 \- \_is\_bail\_out\(\) spots 'Bail out\!  Wing on fire\.'
ok 133 \-   with the right reason
ok 134 \- \_is\_bail\_out\(\) spots 'bail out\! \- Out of coffee'
ok 135 \-   with the right reason
ok 136 \- \_is\_bail\_out\(\) spots 'BAIL OUT\!'
ok 137 \-   with the right reason
ok 138 \- \_is\_bail\_out\(\) spots 'Bail out\!'
ok 139 \-   with the right reason
ok 140 \- \_is\_bail\_out\(\) ignores '  Bail out\!'
ok 141 \-   and gives no reason
ok 142 \- \_is\_bail\_out\(\) ignores 'BAIL OUT'
ok 143 \-   and gives no reason
ok 144 \- \_is\_bail\_out\(\) ignores 'frobnitz'
ok 145 \-   and gives no reason
ok 146 \- \_is\_bail\_out\(\) ignores 'ok 23 \- BAIL OUT\!'
ok 147 \-   and gives no reason
ok

**Comment by p5pRT** [comment]
### From @schwern

On Mon\, Jun 23\, 2003 at 02&#8203;:55&#8203;:15PM \+0200\, Andreas Fehr wrote&#8203;:
> >MGS&#8203;: Or you can run 'make test\_harness'\.
> 
> "Don't know how to make test\_harness"

That's odd\.  Oh\, is this Windows or something?  They're probably missing
that target\.


\-\- 
\.\.\.and that\, children\, is how to clean and load a \.38 revolver\.  Questions?

**Comment by p5pRT** [comment]
### From @steve-m-hay

Resolved&#8203;: perl\-5\.8\.4\+MAINT22869 passes all these tests \(using
MinGW\-3\.1\.0\-1\)\.

**Comment by p5pRT** [comment]
@steve-m-hay - Status changed from 'open' to 'resolved'
